src/pages/PullRequestPage/PullCoverage/routes/FilesChangedTab/FilesChanged/FilesChangedTable/FilesChangedTable.tsx

```diff
@@ -91,48 +91,66 @@ function getColumns({ pullId }: { pullId: string }) {
 
         return (
-          <div className="flex flex-row break-all">
-            <span
-              data-action="clickable"
-              data-testid="file-diff-expand"
-              className={cs(
-                'inline-flex items-center gap-1 font-sans hover:underline focus:ring-2',
-                {
-                  'text-ds-blue': row.getIsExpanded(),
-                }
-              )}
-              onClick={row.getToggleExpandedHandler()}
-            >
-              <Icon
-                size="md"
-                name={row.getIsExpanded() ? 'chevronDown' : 'chevronRight'}
-                variant="solid"
-              />
-            </span>
-            {/* @ts-expect-error */}
-            <A
-              to={{
-                pageName: 'pullFileView',
-                options: { pullId, tree: headName },
-              }}
-            >
-              {headName}
-            </A>
+          <div className="flex flex-row items-center break-all">
+            {!isDeletedFile && (
+              <span
+                data-action="clickable"
+                data-testid="file-diff-expand"
+                className={cs(
+                  'inline-flex items-center gap-1 font-sans hover:underline focus:ring-2',
+                  {
+                    'text-ds-blue': row.getIsExpanded(),
+                  }
+                )}
+                onClick={row.getToggleExpandedHandler()}
+              >
+                <Icon
+                  size="md"
+                  name={row.getIsExpanded() ? 'chevronDown' : 'chevronRight'}
+                  variant="solid"
+                />
+              </span>
+            )}
+            {isDeletedFile ? (
+              <>{headName}</>
+            ) : (
+              /* @ts-expect-error */
+              <A
+                to={{
+                  pageName: 'pullFileView',
+                  options: { pullId, tree: headName },
+                }}
+              >
+                {headName}
+              </A>
+            )}
             {row.original?.isCriticalFile && (
-              <span className="ml-2 rounded border border-ds-gray-tertiary p-1 text-xs text-ds-gray-senary">
-                Critical File
+              <span className="ml-2 h-fit flex-none rounded border border-ds-gray-tertiary p-1 text-xs text-ds-gray-senary">
+                Critical file
               </span>
             )}
+            {isDeletedFile && (
+              <div className="ml-2 h-fit flex-none rounded border border-ds-gray-tertiary p-1 text-xs text-ds-gray-senary">
+                Deleted file
+              </div>
+            )}
           </div>
         )
       },
     }),
     columnHelper.accessor('missesCount', {
       id: 'missedLines',
       header: 'Missed lines',
-      cell: ({ renderValue }) => renderValue(),
+      cell: ({ renderValue, row }) => {
+        if (row.original?.headCoverage === null) {
```

Comment:
Can we check to make sure `headCoverage` exists on `row.original`, in the case where it's `undefined` this case would not be able to pick it up.